crates/sui-core/src/authority.rs

```diff
@@ -748,22 +749,28 @@ impl AuthorityState {
             tx_guard.release();
             return Ok(info);
         }
-
+        let execution_guard = self
+            .database
+            .execution_lock_for_certificate(certificate)
+            .await;
         // Any caller that verifies the signatures on the certificate will have already checked the
         // epoch. But paths that don't verify sigs (e.g. execution from checkpoint, reading from db)
         // present the possibility of an epoch mismatch. If this cert is not finalzied in previous
         // epoch, then it's invalid.
-        if certificate.epoch() != self.epoch() {
+        let Some(execution_guard) = execution_guard else {
```

Comment:
why not just have `execution_lock_for_certificate` return the WrongEpoch error?

Reply:
Good idea, thought it did not reduce code size, because we still need to call this `tx_guard.release`